aem-authoring-toolkit-assets/src/main/content/jcr_root/apps/authoring-toolkit/depends-on/js/dependsOnActionRegistry.js

```diff
@@ -38,12 +39,41 @@
             return action;
         }
 
+        /**
+         * Function to remove symbols that don't match the pattern
+         * @param {string} target - target string
+         * @param {RegExp} pattern -  pattern for correct symbols
+         * */
+        static _replaceForbiddenSymbols(target, pattern) {
+           return target
+               .split('')
+               .map((symbol) => (symbol.search(pattern) !== -1  ? symbol : ''))
+               .join('');
+        }
+
+        /**
+         * Function to develop action's name (allowed symbols a-z, 0-9, -)
+         * @param {string} name - action name
+         * @returns {string} correct action name (according to nameRegex)
+         * */
+        static _getValidName(name) {
+            const nameRegex = /^[a-z0-9-]+$/;
+            let resultName = name;
+            if (name.trim().search(nameRegex) === -1) {
```

Comment:
Searching and sanitizing is almost equal changes.
In case you follow functional practice right you can just check
`if (name !== sanitizedName) console.warn(...)  `
to make a warn message in case of something changed

Reply:
PS: you will not even need this method as operations will be really simple.